### ISSUE TITLE
Allow customers to customize the Channel Type calculation

### ISSUE THREAD

**Issue by robbie-c** [PostHog/posthog#21195]
This issue is to collect votes so that we can understand how important this is. If you're upvoting this issue, please let us know your use case! Either leave a comment below or contact CS.

## Is your feature request related to a problem?

A customer might want to use their own channel type definition, this would let them:
* Add custom channel types
* Change the logic of how channel type is determined
* Add new domains / sources / mediums

## Describe the solution you'd like

Allow any HogQL expression to be used to calculate the channel_type, and allow customers to upload their own channel definitions.

## Describe alternatives you've considered

Do nothing here, maintain one definition for everyone and just add to it or fix it as needed.

## Additional context

Links
* Our definitions list: https://github.com/PostHog/posthog/blob/master/posthog/models/channel_type/channel_definitions.json
* Our channel type hogql expression https://github.com/PostHog/posthog/blob/master/posthog/hogql/database/schema/channel_type.py

#### *Thank you* for your feature request – we love each and every one!

Notes:
* Will need to update the session properties endpoint, to return the custom values, as well as the built-in values.

**Comment by TrevorLoucks** [comment]
Yes, I'm definitely an team allow changing of logic, as mentioned in the support ticket, I use utm_source=x&utm_medium=ppc for X Ads and utm_source=meta&utm_medium=ppc for Meta Ads. I know "Facebook" and "Twitter" are mapped correctly but all my campaigns already run using the new company names.

**Comment by robbie-c** [comment]
Closing as we shipped this :)
